Automattic/wp-calypso
# Build: Add `punycode` to transpiled dependencies

_Merge #33492 and #33496 and rebase this PR before reviewing!_

#### Changes proposed in this Pull Request

Build: Add `punycode` to transpiled dependencies

Part 3 in a series of fixes for for https://github.com/Automattic/jetpack/issues/12511. Quoting part 1 (#33492):

> I would like to move [...] logic that's taking care of transpiling certain `node_modules` into the `calypso-build` package. The reason for this is that Jetpack (which uses `calypso-build` in its toolchain) has currently [a number of Gutenberg blocks that are broken in IE11](https://github.com/Automattic/jetpack/issues/12511) because of a 3rd party dependency that ships untranspiled code. See https://github.com/Automattic/jetpack/issues/12511#issuecomment-497593885 for how I would like to fix this.

#### Testing instructions

```
npm run distclean && npm ci
npm start
```

Verify that the app still builds and runs. In particular, check sections that involve any of the transpiled dependencies (Stats screen for `d3`) still work.

_Equally important:_ `npm link` Jetpack against the `@automattic/calypso-build` package on this branch, and rebuild it. Test it in IE11 to make sure https://github.com/Automattic/jetpack/issues/12511 is fixed.

#### Follow-up

Release the new version, and file a Jetpack PR to use it.
